### PR TITLE
Write blog post about low-memory indexing with bloom filters

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: website/blog/2019-01-22-bloom-filters.md [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+
+```scala
+val references: Map[Path, BloomFilter[Symbol]]
+```
+
+The values represent the set of
+[SemanticDB symbols](https://scalameta.org/docs/semanticdb/specification.html)
+referenced in that file. A nice property of keying the map by file paths is that
+we can incrementally update the map as files change. When a file is re-compiled
+producing a new SemanticDB file, we throw out the old bloom filter a compute a
```
Comment (marcelocenerine) [inline website/blog/2019-01-22-bloom-filters.md]: a[nd] compute

File: website/blog/2019-01-22-bloom-filters.md [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+- when creating a bloom filter, you must provide an estimate for how many
+  elements will be added to the set. A large estimate results in lower false
+  positive rates for the `contains(element)` method at the cost of higher space
+  usage. Conversely, a low estimate results in lower memory usage at the cost of
+  higher false positives for the `contains(element)` method.
+- you can't iterate through the elements of a bloom filter.
+
+In exchange for these limitations, bloom filters are able to compress a large
+number of elements into a small number of bits. Due to their space-efficiency,
+bloom filters are used in a many applications ranging from browsers, CDNs and
```
Comment (marcelocenerine) [inline website/blog/2019-01-22-bloom-filters.md]: in ~a~ many

File: website/blog/2019-01-22-bloom-filters.md [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+whenever Metals starts in an existing project or after the build changes. Can
+indexing be made faster? Probably yes, but the user experience will still remain
+bottle-necked by sbt build export and compilation of workspace sources, which
+frequently take many minutes to complete on large projects.
+
+The indexes are in-memory maps where the keys are file paths and values are
+bloom filters. When files change, we can incrementally update the indexes by
+computing a new bloom filter for the updated source file.
+
+The bloom filter indexes are only used to narrow down the search state by
```
Comment (marcelocenerine) [inline website/blog/2019-01-22-bloom-filters.md]: search ~state~ space?

File: docs/editors/sublime.md [hunk 4]
```diff
@@ -79,19 +85,41 @@ Once the import step completes, compilation starts for your open `*.scala`
...
+The default key binding is `shift+F12`. If you use vim-binding, you need to be
+in insert-mode.
+
+![Find references](https://i.imgur.com/BJDkczD.gif)
+
 ## Manually trigger build import
-When you change build.sbt or sources under project/ you you will need to 
-to manually trigger a build import, for this you can register a custom sublime command.More about command pallet [here](http://docs.sublimetext.info/en/latest/reference/command_palette.html).
+
+When you change build.sbt or sources under project/ you you will need to to
```
Comment (gabro) [inline docs/editors/sublime.md]: double "you"

Comment (olafurpg) [reply docs/editors/sublime.md]: fixed, thanks!

**Pull request merged**